DemocracyClub/EveryElection
# Feature/add id requirements

This work aims to add a field, `requires_voter_id` to the elections database, populate the existing records and ensure that any new elections created via the ID creator have their ID requirements assigned correctly. This is the preliminary work for doing the ID messaging ([Trello](https://trello.com/c/UvMnJkuS/3207-voter-id-messaging)).  Closes #1729 

Also, added a nice dropdown list to the admin panel where ID requirements can be manually assigned/overridden.
<img width="511" alt="Screenshot 2023-03-09 at 16 04 10" src="https://user-images.githubusercontent.com/9531063/224082438-d504230f-abd3-4fc5-a13f-2d574b9eaf61.png">

Due to the election ID relying on divisions having territory codes, this PR is blocked by #1865 for the moment.


To test this locally:
- Make sure you've done a fresh requirements install
- Run the migration
- Check your elections table for anything that you know should need ID (I'd suggest a local election in ENG/WAL/SCT in May 2023) and check that the `requires_voter_id` column is set to `EA-2022` (n.b. it will be `EFA-2002` if you're checking a NI election)
- Run EE locally and use the ID creator to make any GB parliamentary election in November for any constituency you like
- Find the new election in your elections table and verify that it has `requires_voter_id` set to `EA-2022`
- Log in to your local admin panel, find the election you created, and set the voter id requirements to something else using the dropdown 
- Save it and check it's updated in your db.


```[tasklist]
### PR Checklist

- [x] No unintentional console.logs left behind after debugging
- [x] Did I add or updated any dependencies? Explain why.
```
